code/modules/crew_objectives/engineering_objectives.dm

```diff
@@ -32,3 +32,41 @@
 			if(istype(dumbbird.ears, /obj/item/radio/headset))
 				return TRUE
 	return ..()
+
+/datum/objective/crew/supermatter_survive
+	explanation_text = "Prevent the primary Supermatter engine from fully delaminating. Additional Supermatter shards ordered from cargo do not count for this"
+	jobs = "chiefengineer,stationengineer"
+
+/datum/objective/crew/supermatter_survive/check_completion()
+	if(GLOB.main_supermatter_engine)  // handy it qdel's itself
+		return TRUE
+	return ..()
+
+
+/datum/objective/crew/apc
+	explanation_text = "Make sure the station has above (something broke message us)kW in the powernet."
+	jobs = "chiefengineer,stationengineer,atmospherictechnician"
+
+/datum/objective/crew/apc/New()
+	. = ..()
+	target_amount = rand(100,500)  // no i dont know how much a normal amount is :>
+	explanation_text = "Make sure the station has above [target_amount]kW in the powernet."
+
+/datum/objective/crew/apc/check_completion()
+	var/total_power = 0
+	var/list/powermonitors = list()
+
+	for(var/obj/machinery/computer/monitor/pMon in GLOB.machines)
+		if(pMon.stat & (NOPOWER | BROKEN)) //check to make sure the computer is functional
+			continue
+		if(pMon.is_secret_monitor) //make sure it isn't a secret one (ie located on a ruin), allowing people to metagame that the location exists
+			continue
+		powermonitors += pMon
+
+	for(var/obj/machinery/computer/monitor/pMon in powermonitors)
+		var/datum/powernet/connected_powernet = pMon.get_powernet()
+		total_power += connected_powernet.viewavail
```

Comment:
This doesn't work at all, multiple power nets on the same grid will add the same powernets twice.
Just loop through all powernets in the world, its a good enough solution

Reply:
so... just replace it with for(obj/whatever/powernet in world?)

Reply:
I dont know, in world is usually pretty bad. Find a good way to get all the powernets (The power subsystem?)